src/ucs/datastruct/string_buffer.c

```diff
@@ -125,6 +125,18 @@ void ucs_string_buffer_append_flags(ucs_string_buffer_t *strb, uint64_t mask,
     ucs_string_buffer_rtrim(strb, ",|");
 }
 
+void ucs_string_buffer_append_iovec(ucs_string_buffer_t *strb,
+                                    const struct iovec *iov, size_t iovcnt)
+{
+    size_t iov_index;
+
+    for (iov_index = 0; iov_index < iovcnt; ++iov_index) {
+        ucs_string_buffer_appendf(strb, "%p,%zu|", iov[iov_index].iov_base,
+                                  iov[iov_index].iov_len);
+    }
+    ucs_string_buffer_rtrim(strb, "|");
```

Comment:
in case if you call this function twice you get unreadable string like
```0x12345678,1230x56781234,456```

Reply:
yes, the user is responsible to insert separators in between if needed